orcid-message-listener/src/main/java/org/orcid/listener/orcid/Orcid20APIClient.java

```diff
@@ -60,11 +65,17 @@ public Orcid20APIClient(@Value("${org.orcid.message-listener.api20BaseURI}") Str
     /**
      * Fetches the public record from the ORCID API v2.0
      * 
+     * Caches based on message.
+     * 
      * @param orcid
      * @return Record
      */
-    public Record fetchPublicRecord(String orcid) throws LockedRecordException, DeprecatedRecordException {
-        WebResource webResource = jerseyClient.resource(baseUri).path(orcid + "/record");
+    public Record fetchPublicRecord(BaseMessage message) throws LockedRecordException, DeprecatedRecordException {
+        Record cached = v2PerMessageQueue.getIfPresent(message);
```

Comment:
Hey @TomDemeranville, 

Shouldn't we implement hashcode and equals on the BaseMessage? if not, how the cache ensures that the given message is exactly the same as the one cached?

Thanks

Reply:
Ah, I think it's abstract, so it would be the implementations that get called, right?  I will add them to RetryMessage and LastModifiedMessage

Reply:
Actually, you're right.  Both subclasses have all their state in the BaseMessage map.  I'll put the equals there.